ctru-rs/src/services/ac.rs

```diff
@@ -0,0 +1,349 @@
+    #[doc(alias = "ACU_GetProxyEnable")]
+    pub fn get_proxy_enabled(&self) -> crate::Result<bool> {
+        unsafe {
+            let mut ret = false;
+            ResultCode(ctru_sys::ACU_GetProxyEnable(&mut ret))?;
+
+            Ok(ret)
+        }
+    }
+
+    /// Returns the connected network's proxy port, if present.
+    ///
+    /// You can check if the console is using a proxy with [`Ac::get_proxy_enabled()`]
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # let _runner = test_runner::GdbRunner::default();
+    /// # use std::error::Error;
+    /// # fn main() -> Result<(), Box<dyn Error>> {
+    /// #
+    /// use ctru::services::ac::Ac;
+    ///
+    /// let ac = Ac::new()?;
+    ///
+    /// println!("Proxy port: {}", ac.get_proxy_port()?);
+    /// #
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[doc(alias = "ACU_GetProxyPort")]
+    pub fn get_proxy_port(&self) -> crate::Result<u32> {
+        unsafe {
+            let mut ret = 0u32;
+            ResultCode(ctru_sys::ACU_GetProxyPort(&mut ret))?;
+
+            Ok(ret)
+        }
+    }
+
+    /// Returns the connected network's proxy username, if present.
+    ///
+    /// You can check if the console is using a proxy with [`Ac::get_proxy_enabled()`]
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # let _runner = test_runner::GdbRunner::default();
+    /// # use std::error::Error;
+    /// # fn main() -> Result<(), Box<dyn Error>> {
+    /// #
+    /// use ctru::services::ac::Ac;
+    ///
+    /// let ac = Ac::new()?;
+    ///
+    /// println!("Proxy username: {}", ac.get_proxy_username()?);
+    ///
+    /// #
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[doc(alias = "ACU_GetProxyUserName")]
+    pub fn get_proxy_username(&self) -> crate::Result<String> {
+        unsafe {
+            let mut vec = vec![0u8; 0x20];
+            ResultCode(ctru_sys::ACU_GetProxyUserName(vec.as_mut_ptr()))?;
+
+            Ok(String::from_utf8(vec)?)
+        }
+    }
+
+    /// Returns the connected network's proxy password, if present.
+    ///
+    /// You can check if the console is using a proxy with [`Ac::get_proxy_enabled()`]
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # let _runner = test_runner::GdbRunner::default();
+    /// # use std::error::Error;
+    /// # fn main() -> Result<(), Box<dyn Error>> {
+    /// #
+    /// use ctru::services::ac::Ac;
+    ///
+    /// let ac = Ac::new()?;
+    ///
+    /// println!("Proxy password: {}", ac.get_proxy_password()?);
+    /// #
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[doc(alias = "ACU_GetProxyPassword")]
+    pub fn get_proxy_password(&self) -> crate::Result<String> {
+        unsafe {
+            let mut vec = vec![0u8; 0x20];
+            ResultCode(ctru_sys::ACU_GetProxyPassword(vec.as_mut_ptr()))?;
+
+            Ok(String::from_utf8(vec)?)
+        }
+    }
+
+    /// Load the selected network slot, if present.
+    ///
+    /// Note: this method requires `ac:i` access
```

Comment:
Is this access given by the normal `Ac::new()` implementation, or should the user do something more to gain it? Reimplementing IPS calls isn't exactly a straightforward process.

Reply:
I think normally Citra grants basically _all_ permissions since it's just emulated, although it's possible some methods are just unimplemented / stubs, I don't think permissions are usually denied.